### PR TITLE
Avoid useless erasure of AqlValues when a block is being destroyed

### PR DESCRIPTION
**Pull request by goedderz**
Minor change, avoid some useless work.

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10339/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10342/

File: arangod/Aql/AqlItemBlock.cpp [hunk 1]
```diff
@@ -295,34 +295,23 @@ void AqlItemBlock::destroy() noexcept {
...
-            _valueCount.erase(it2);
-          }
-        }
+    size_t totalUsed = 0;
+    for (auto& it : _valueCount) {
+      auto const& count = it.second;
+      if (ADB_LIKELY(count > 0)) {
+        auto value = it.first;
+        value.destroy();
+        totalUsed += value.memoryUsage();
```
Comment (jsteemann) [inline arangod/Aql/AqlItemBlock.cpp:304]: shouldn't these two lines be inversed? how can we tell the `value`'s memoryUsage otherwise?

Comment (goedderz) [reply arangod/Aql/AqlItemBlock.cpp:304]: 🤦

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10345/

**Pull request closed without merging**